torchchat/cli/builder.py

```diff
@@ -598,6 +606,117 @@ def do_nothing(max_batch_size, max_seq_length):
             model = PTEModel(config, builder_args.pte_path)
         except Exception:
             raise RuntimeError(f"Failed to load ET compiled {builder_args.pte_path}")
+    elif builder_args.distributed:
+        # Using params_table to identify the model to load, for example "Meta-Llama-3.1-8B".
+        #TODO This is a hacky way to please the distributed loading api and needs to be replaced
+        NAME_TO_DISTRIBUTION = {
+            "Meta-Llama-3-8B": "meta-llama/Meta-Llama-3-8B-Instruct", 
+            "Meta-Llama-3.1-8B": "meta-llama/Meta-Llama-3.1-8B-Instruct",
+            "Meta-Llama-3-70B": "meta-llama/Meta-Llama-3-70B-Instruct",
+            "Meta-Llama-3.1-70B": "meta-llama/Meta-Llama-3.1-70B-Instruct",
+
+        }
```

Comment:
What's the gap to using the models described in `model_config/models.json`.  

Definitely should not be part of the present PR, but I think as a north star, it would be desirable to grab the same models (and download and mgmt infra etc) for non-distributed and distributed.

Reply:
Thanks for the comment @mikekgfb. The gap wasn't that big, the args.model just wasn't accessible at that point and I wanted to take a deeper look to fix it right. Removed the cruel hack and I now save the distribution_path when creating the builder_arg. Still not sure if this is the intended way of locating the checkpoint though.

Reply:
Not sure what you're trying to do.  If it's just that the dict/json data structure describing the mapping isn't in scope, maybe what you want is some methods that give you the relevant info?

Also, these seem to be mapping of short names to HF network paths - should we not have a way to [ick them from the local filesystem (since the torchchat cli already manages download and all that).  Oh, and if the answer is "we have bigger fish to fry, hooking this up is not highest priority" I will wholeheartedly agree.  This is more about understanding the context of this PR.  

Where I'm lacking the context is how you go from all the weights being available locally on a node to reading those weights on another node?  And maybe that's why you prefer to straight up pick the files from HF?  (Although local distribution from an already downloaded set of weights probs has higher bandwidth?)  Again, there's much bigger fish to fry, and I think this PR is a good step in the direction of frying those fish ;)

Reply:
Oh I see, let me try to fill in some context for this PR. Previously, the distributed inference solution lived completely separate in its own script (within the torchchat repo but) separate from torchchat.py. Distributed [provides its own utils](https://github.com/pytorch/torchchat/blob/edc2cfbd81b54f60a31239588007bba619f91fd7/torchchat/distributed/checkpoint_utils.py) to load either hf or torchchat weights (where the torchchat part is currently broken IIRC). In a previous PR (#1327), I enabled the usage of `torchchat.py generate` with a distributed model. This PR only progresses the integration into the cli by enabling chat/server but stops short from replacing the weight loading mechanics which are still custom to distributed.

So, yes, for this PR I was only looking for a quick and dirty way to map arg.model_name -> "huggingface distribution str" (without actually having model_name at hand) to load the weights from the [hf cache](https://github.com/pytorch/torchchat/blob/edc2cfbd81b54f60a31239588007bba619f91fd7/torchchat/distributed/checkpoint_utils.py#L67). I now modified the PR to use the information provided in the model_config/models.json as you suggested. In a next PR we should then alter `torchchat/distributed/checkpoint_utils.py` to leverage torchchat infra like (e.g. builder_args) to locate and access the files.